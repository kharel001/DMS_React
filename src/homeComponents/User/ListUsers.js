import React, { useEffect} from 'react';
import MaterialTable from 'material-table';
import TableIcons from '../../homeComponents/TableIcons/TableIcons';
import { useSelector, useDispatch } from 'react-redux'
import * as actions from '../../store/actions/users'
import { Redirect } from 'react-router';

/**
* A List of healthpost users
*
* @param - No parameters
* @returns {ListUsers} - returns a material table of healthpost users
*/

const ListUsers = () => {

    const [state, setState] = React.useState({
        columns: [
            { title: 'First Name', field: 'firstname' },
            { title: 'Last Name', field: 'lastname' },
            { title: 'Health Post', field: 'healthPost' },
            { title: 'Address', field: 'address' },
            { title: 'Phone Number', field: 'phonenumber' },
            { title: 'Email', field: 'email' },
            { title: 'Password', field: 'password' }
        ],
        data: [
        ],
    });
    const dispatch = useDispatch();
    const userData = useSelector(({ users }) => users.users)
    const [redirectTo, setRedirectTo] = React.useState(null);

    useEffect(() => {
        dispatch(actions.getUsers());
    }, [dispatch])

    useEffect(() => {
        setState(prevState => { return { ...prevState, data: userData.healthpost } })
    }, [userData])

    const [selectedRow, setSelectedRow] = React.useState(null);

    const selectUser = (id) => {
        // console.log(id);
        setRedirectTo(<Redirect to={{
            pathname: '/app/users/profile',
            state: { id: id }
        }} />)
    }
    // console.log(userData);
    return (
        <div className="app-wrapper">
            {redirectTo}
            <div className="animated slideInUpTiny animation-duration-3">
                <MaterialTable
                    title="Users"
                    columns={state.columns}
                    data={state.data}
                    icons={TableIcons}
                    onRowClick={((evt, selectedRow) => selectUser(selectedRow.tableData.id))}
                    options={{
                        rowStyle: rowData => ({
                            backgroundColor: (selectedRow === rowData.tableData.id) ? '#EEE' : '#FFF',
                            font: 'Roboto'
                        }),
                        headerStyle: {
                            backgroundColor: '#01579b',
                            color: '#FFF'
                        }
                    }}
                    editable={{
                        onRowAdd: (newData) =>
                            new Promise((resolve) => {
                                setTimeout(() => {
                                    resolve();
                                    console.log(newData);
                                    //   dispatch(actions.addMedicine(newData))
                                    setState((prevState) => {
                                        const data = [...prevState.data];
                                        data.push(newData);
                                        return { ...prevState, data };
                                    });
                                }, 600);
                            }),
                        onRowUpdate: (newData, oldData) =>
                            new Promise((resolve) => {
                                setTimeout(() => {
                                    resolve();
                                    if (oldData) {
                                        console.log(newData);
                                        // dispatch(actions.updateMedicine(newData, newData._id))
                                        setState((prevState) => {
                                            const data = [...prevState.data];
                                            data[data.indexOf(oldData)] = newData;
                                            return { ...prevState, data };
                                        });
                                    }
                                }, 600);
                            }),
                        onRowDelete: (oldData) =>
                            new Promise((resolve) => {
                                setTimeout(() => {
                                    resolve();
                                    setState((prevState) => {
                                        const data = [...prevState.data];
                                        data.splice(data.indexOf(oldData), 1);
                                        // dispatch(actions.deleteMedicine(oldData._id))
                                        return { ...prevState, data };
                                    });
                                }, 600);
                            }),
                    }}
                />
            </div>
        </div>)
}

export default ListUsers;