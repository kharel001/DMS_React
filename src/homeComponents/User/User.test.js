import React from 'react';
import User from './User';
import {
    configure,
    mount,
} from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import {act} from 'react-dom/test-utils'

const mockStore = configureStore([]);
// JestHook.mock('expo-font');
configure({
    adapter: new EnzymeAdapter
})

const changedFunction=jest.fn();
const dummy_data = {
  title:"title",
  users:[
      {
          username:"sa"
      },
      {
        username:"ga"
    },
    {
        username:"la"

    }
  ]
}

const store = mockStore({
    });


const setup = (props = {}) => {
    return mount(<Provider store={store}><MemoryRouter>< User
     {...props}  /></MemoryRouter></Provider>
    )
}

const findByTestAttr=(wrapper, val)=>{
    return wrapper.find(`[data-test='${val}']`)
}

{/* it("Custom Scrollable render", () => {
        let wrapper = setup(dummy_data);
        // expect(wrapper).toMatchSnapshot();
}) */}
it("Snapshot User", () => {
    let wrapper = setup(dummy_data);
    expect(wrapper).toMatchSnapshot();
   
})



it("Components User", () => {
    let wrapper = setup(dummy_data);
    let titleComp = findByTestAttr(wrapper, "titleComp").at(0);
    expect(titleComp.text()).toEqual("title")
    let listItemComp = findByTestAttr(wrapper, "listItemComp").at(3);
    expect(listItemComp.prop('primary')).toEqual("ga")
    let iconButtonComp = findByTestAttr(wrapper, "iconButtonComp").at(3);
    // expect(iconButtonComp.prop('primary')).toEqual("sa")
    let modalComp=findByTestAttr(wrapper,"modalComp").at(0)

    expect(modalComp.prop('open')).toBe(false)
    expect(modalComp.prop('user')).toEqual({})

    act(()=>{iconButtonComp.props().onClick()})
    wrapper.update()
    let modalComp1=findByTestAttr(wrapper,"modalComp").at(0)

    expect(modalComp1.prop('open')).toBe(true)
    expect(modalComp1.prop('user')).toEqual( {
        username:"sa"
    },)
    act(()=>{modalComp1.props().onClose()})
    wrapper.update()
    let modalComp2=findByTestAttr(wrapper,"modalComp").at(0)

    expect(modalComp2.prop('open')).toBe(false)  

})
