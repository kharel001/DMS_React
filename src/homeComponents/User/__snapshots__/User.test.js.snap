// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot User 1`] = `ReactWrapper {}`;
